hermes/pkg/hermes/hermes.go

```diff
@@ -0,0 +1,400 @@
+package hermes
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/exec"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/step"
+	"github.com/ignite/cli/ignite/pkg/localfs"
+	"github.com/ignite/ignite-files/hermes"
+)
+
+const (
+	FlagHostChain        = "host-chain"
+	FlagReferenceChain   = "reference-chain"
+	FlagChainA           = "a-chain"
+	FlagChainB           = "b-chain"
+	FlagClientA          = "a-client"
+	FlagClientB          = "b-client"
+	FlagConnectionA      = "a-connection"
+	FlagConnectionB      = "b-connection"
+	FlagPortA            = "a-port"
+	FlagPortB            = "b-port"
+	FlagShowCounterparty = "show-counterparty"
+	FlagChain            = "chain"
+	FlagMnemonicFile     = "mnemonic-file"
+	FlagConfig           = "config"
+	FlagJSON             = "json"
+)
+
+const (
+	// CommandCreate is the Hermes create command.
+	cmdCreate cmdName = "create"
+
+	// CommandQuery is the Hermes query command.
+	cmdQuery cmdName = "query"
+
+	// CommandKeys is the Hermes keys command.
+	cmdKeys cmdName = "keys"
+
+	// CommandStart is the Hermes start command.
+	cmdStart cmdName = "start"
+
+	// CommandClient is the Hermes create client command.
+	cmdClient subCmd = "client"
+
+	// CommandConnection is the Hermes create connection command.
+	cmdConnection subCmd = "connection"
+
+	// CommandChannel is the Hermes create channel command.
+	cmdChannel subCmd = "channel"
+
+	// CommandChannels  is the Hermes query channels command.
+	cmdChannels subCmd = "channels"
+
+	// CommandKeysAdd is the Hermes keys add command.
+	cmdKeysAdd subCmd = "add"
+
+	// cmdKeysList is the Hermes keys list command.
+	cmdKeysList subCmd = "list"
+
+	// ResultSuccess is the api result status success.
+	ResultSuccess = "success"
+
+	// ResultError is the api result status error.
+	ResultError = "error"
+)
+
+type (
+	// Flags represents the Hermes run flags.
+	Flags map[string]interface{}
+	// cmdName represents a high level command under Hermes.
+	cmdName string
+	// SubCommand represents the sub command under Hermes.
+	subCmd string
+
+	// Hermes represents the hermes binary structure.
+	Hermes struct {
+		path    string
+		binary  []byte
+		cleanup func()
+	}
+
+	// Option configures Generate configs.
+	Option func(*configs)
+
+	// configs holds Generate configs.
+	configs struct {
+		flags  Flags
+		config string
+		stdOut io.Writer
+		stdErr io.Writer
```

Comment:
Maybe, to keep consistency with existing code:

```suggestion
		stdout io.Writer
		stderr io.Writer
```